doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,481 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  *********************************************************************/
 
-/* Author: Sachin Chitta, Dave Coleman, Mike Lautman */
+/* Author: Sachin Chitta, Dave Coleman, Mike Lautman, Simon Goldstein */
 
+// In C++ to use the move_group_interface it and several other files must be included.
+// Both the move_group and planning_scene are accessed through their appropriate interface.
 #include <moveit/move_group_interface/move_group_interface.h>
 #include <moveit/planning_scene_interface/planning_scene_interface.h>
-
+// We also include visual_tools and several other useful message types.
+#include <moveit_visual_tools/moveit_visual_tools.h>
 #include <moveit_msgs/DisplayRobotState.h>
 #include <moveit_msgs/DisplayTrajectory.h>
-
 #include <moveit_msgs/AttachedCollisionObject.h>
 #include <moveit_msgs/CollisionObject.h>
 
-#include <moveit_visual_tools/moveit_visual_tools.h>
+// The rviz_vizual_tools namespace is shortened for code readability.
+namespace rvt = rviz_visual_tools;
 
-int main(int argc, char** argv)
+class MoveGroupInterfaceTutorial
 {
-  ros::init(argc, argv, "move_group_interface_tutorial");
-  ros::NodeHandle node_handle;
-  ros::AsyncSpinner spinner(1);
-  spinner.start();
-
-  // BEGIN_TUTORIAL
+public:
+
+  /**
+  * Basic constucter
+  **/
+  MoveGroupInterfaceTutorial()
+  {
+  }
+  /**
+  * This initializes visual_tools_ and sets the text_pose_ to above the robot
+  */
+  void setupVisualization()
+  {
+    // BEGIN_SUB_TUTORIAL viz_setup
+    //
+    // Visualization
+    // ^^^^^^^^^^^^^
+    //
+    // Within MoveItVisualTools, remote control is an introspection tool that allows users to
+    // step through a high level script via buttons and keyboard shortcuts in RViz.
+    // A variety of commands can be used to display anything from text, points or even paths.
+    //
+    visual_tools_.loadRemoteControl();
+    // END_SUB_TUTORIAL
+
+    // Set the location where text is diplayed to above the robot
+    text_pose_.translation().z() = 1.25;
+  }
+  /**
+  * Publishes the given message above the robot
+  **/
+  void publishToRViz(std::string msg)
```

Comment:
add an optional argument `bool blocking=false` and only call prompt if `blocking == true`

Reply:
I set it to default to true because I felt it resembled the average use of the function.